src/WebJobs.Extensions.Http/ClaimsPrincipalHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Runtime.Serialization;
+    using System.Runtime.Serialization.Json;
+    using System.Security.Claims;
+
+    internal static class ClaimsPrincipalHelper
+    {
+        // Shared serializer instance. This is safe for multi-threaded use.
+        private static readonly DataContractJsonSerializer Serializer = GetCustomSerializer();
+
+        public static ClaimsPrincipal FromBindingData(IReadOnlyDictionary<string, object> bindingData)
+        {
+            var req = bindingData.Values.FirstOrDefault(val => val.GetType() == typeof(HttpRequestMessage)) as HttpRequestMessage;
```

Comment:
>HttpRequestMessage [](start = 87, length = 18)

Does functions v2 also use HttpRequestMessage internally (and not the new HttpRequest in .NET Core)?

Reply:
Good point. I believe in functions v2, either HttpRequestMessage or HttpRequest can be utilized as this type. However, this current PR is for function v1 (which uses webjobs v2), and targets .NET 4.6, so I don't think I even have access to this type.

---
In reply to: [174631976](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/373#discussion_r174631976) [](ancestors = 174631976)